### ISSUE TITLE
Huge GL-Loss values on a text data encoded with word embedding.

### ISSUE THREAD

**Issue by moeinsh78** [jiangboahu/GLCN-tf#15]
Hi, 
I have been working with your implemented model for text classification tasks. However, I have encountered an issue related to the calculation of the Graph Learning loss when using pre-trained word embeddings.

I have been using your model on a textual dataset represented as a graph for text classification. Initially, I experimented with a diagonal matrix for each graph node as an initial embedding (one-hot initial embeddings), and the model exhibited reasonable behavior. The loss value started from a value near 2 and progressively decreased in subsequent epochs.

However, when I attempted to utilize pre-trained word embeddings such as GloVe or Fasttext to encode my graph nodes, I observed abnormal behavior in the Graph Learning loss calculation. In the first epoch, the loss value rose to unusually large values, around 200,000. Although it began to gradually decrease, the learning process seemed to be compromised, as the accuracies remained stagnant throughout all training epochs, hovering around 60%.

Furthermore, I experimented by rescaling the pre-trained word embeddings' values to the [0, 1] range. This adjustment caused the loss value to start from 2000 and decrease, but the core issue with the learning process and static accuracy persisted.

At this point, I suspect there might be an issue with the graph learning loss calculator in the provided Python code. 
D = tf.linalg.matmul(tf.transpose(self.x), D)
After this line of code, the D output contains values around 200 in the first epoch, even when the embeddings’ values are scaled between 0 and 1, which seems to be abnormal.

I would greatly appreciate any insights, suggestions, or assistance you can provide to help resolve this issue. 
Thank you for your attention and support. 
Best regards
